### PR TITLE
samples: nrf_rpc: protocol_serialization: add nfc

### PR DESCRIPTION
**Pull request by alxelax**
PR fixes the sequence of the serialized parameters in event callback serializer to make compatible client and server parts.
PR adds NFC API usage in the protocol serialization samples (both client and server) as well as extends samples documentation.

### CODE REVIEW

**Review by wiba-nordic — APPROVED** [review]
Since the samples are new to the latest release, changes to them don't need a changelog entry.

**Comment by alxelax** [comment]
just rebased to pass CI

**Review by Damian-Nordic — APPROVED** [review]

File: samples/nrf_rpc/protocols_serialization/server/README.rst [hunk 1]
```diff
@@ -312,9 +315,54 @@ Complete the following steps to test OpenThread API serialization:
...
+   This indicates that the communication between the devices has been initialized properly.
+
+#. Run the following command on the client's terminal emulator to bring up the OpenThread interface on the server device:
+
+   .. code-block:: console
+
+      uart:~$ nfc init
+      uart:~$ nfc start
+
+#. Start the any application on smartphone or tablet that is able to read NFC tags.
```
[the remark this answers is not in the document]
Comment (alxelax) [reply samples/nrf_rpc/protocols_serialization/server/README.rst:341]: I agree with `the`. However, I'm not sure about `your`. Will it work if I use not mine but smartphone of my wife? :) 
I understand the question is artificial and quite stupid. Nevertheless.

[the remark this answers is not in the document]
Comment (Damian-Nordic) [reply samples/nrf_rpc/protocols_serialization/server/README.rst]: ok, skip `your` :)

**Review by canisLupus1313 — APPROVED** [review]

**Pull request merged**